### ISSUE TITLE
run_metric should bail out early when an invalid metric is passed.

### ISSUE THREAD

**Issue by natduca** [catapult-project/catapult#2200]
I went to try to use run_metric yesterday. Here's what happened:

```
  884  [2016-03-31 16:02:43 -0700] ./tracing/bin/run_metric
  885  [2016-03-31 16:03:32 -0700] ./tracing/bin/run_metric EfficiencyMetric ~/Downloads/trace_bruce.json
  886  [2016-03-31 16:03:39 -0700] ./tracing/bin/run_metric EfficiencyMetric ~/Downloads/trace_mostly_good_ytp.json
  887  [2016-03-31 16:03:52 -0700] ./tracing/bin/run_metric efficiency_metric ~/Downloads/trace_mostly_good_ytp.json
  888  [2016-03-31 16:04:06 -0700] ./tracing/bin/run_metric system_health/efficiency_metric ~/Downloads/trace_mostly_good_ytp.json
```

First I read the help. That was cool. I tried the metric name and a trace. It printed a huge json blob that after I squinted enough looked like an error. I assumed the trace was busted. I changed the trace. Same error. Only then was it obvious that the metric was specified wrong.

So I tried efficiency_metric. Nope.

Oh, how about system_health/efficiency_metric. Yep.

Why no html suffix? Seems to work either way. Totally nonobvious.

And, the message printed is super hard to grok. Can we haz clearer failure message when its the loading and not the metric exploding?

**Comment by eakuefner** [comment]
1. I agree that we should handle deserializing failures and printing the stack trace legibly.
2. If you're able to pass a filename for the metric and it works, you probably need to rebase. For what it's worth, I can't repro that at ToT and @lpy changed this in #2098 so that what you pass is the name of a registered metric (since we introduced `all_metrics.html`.

We should update the help text to make that very clear, though.

**Comment by eakuefner** [comment]
We should use the machinery which will be used to fix #2159 to fix this bug too.
